chore(APP-207): Migrate from Yarn to pnpm for package management

tyhonchik opened the pull request:
# Description

- Updated package manager from Yarn to pnpm across the project.
- Added .npmrc configuration for pnpm settings.
- Removed Yarn-specific files (.yarnrc.yml, yarn.lock, .nvmrc).
- Updated GitHub Actions workflows to use pnpm for dependency installation and scripts.
- Adjusted lint-staged configuration to work with pnpm.
- Updated README to reflect changes in setup instructions and benefits of using pnpm.

### 📊 Performance Improvements

| Metric | Before (Yarn 4.9.2) | After (pnpm 10.18.3) | Improvement |
|--------|---------------------|----------------------|-------------|
| **Cold Install** | 37.5s | 10.8s | **71% faster** |
| **Warm Install** | 25.3s | 2.5s | **90% faster** |
| **node_modules Size** | 1.8GB | 1.4GB | **22% smaller** |

### Changes Made

- **Package Manager**: Migrated from Yarn 4.9.2 to pnpm 10.18.3
- **Configuration**: Updated `.npmrc` with secure pnpm settings
- **Dependencies**: Fixed phantom dependencies and type declarations
- **Build**: Resolved MetaMask SDK React Native compatibility issues
- **Node.js**: Configured automatic version management via pnpm

### Security Enhancements

- `save-exact=true` - Prevents version drift
- `minimum-release-age=10080` - 7-day freeze on new packages
- `ignore-scripts=true` - Prevents malicious script execution
- `strict-peer-dependencies=true` - Enforces dependency contracts

### Dependency Management

- **Phantom Dependencies Fixed**: `@radix-ui/react-toggle-group`, `@radix-ui/react-dialog`, `@sentry/core`, `@walletconnect/types`
- **Type Safety**: Added `auto-install-types=true` for automatic `@types/*` installation
- **Node Version**: Automatic management via `use-node-version=22.20.0`

*Measured on Aragon App project during migration (Oct 2025)*

## Type of Change

<!--- Please delete the options that are not relevant. -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Developer Checklist:

- [x] Manually smoke tested the functionality locally
- [x] Confirmed there are no new warnings or errors in the browser console
- [ ] (For User Stories only) Double-checked that all Acceptance Criteria are satisfied
- [x] Confirmed there are no new warnings on automated tests
- [x] Merged and published any dependent changes in downstream modules
- [x] Selected the correct base branch
- [x] Commented the code in hard-to-understand areas
- [x] Followed the code style guidelines of this project
- [x] Reviewed that the Files Changed in Github’s UI reflect my intended changes
- [ ] Confirmed the pipeline checks are not failing

## Review Checklist:

- [ ] Tested locally that all Acceptance Criteria or Expected Outcomes are satisfied
- [ ] Confirmed that changes follow the code style guidelines of this project

sepehr2github reviewed (approved):
Thanks Pieter, Looks good to go

tyhonchik committed:
chore(APP-207): Migrate from Yarn to pnpm for package management

- Updated package manager from Yarn to pnpm across the project.
- Added .npmrc configuration for pnpm settings.
- Removed Yarn-specific files (.yarnrc.yml, yarn.lock, .nvmrc).
- Updated GitHub Actions workflows to use pnpm for dependency installation and scripts.
- Adjusted lint-staged configuration to work with pnpm.
- Updated README to reflect changes in setup instructions and benefits of using pnpm.

tyhonchik committed:
chore: Update dependencies and configuration for pnpm

- Updated pnpm version to 10.19.0 in package.json and .npmrc.
- Set auto-install-peers to false in .npmrc for better control over peer dependencies.
- Upgraded @reown/appkit and @reown/appkit-adapter-wagmi to version 1.8.10.
- Updated wagmi to version 2.18.1.
- Added webpack as a new dependency with version 5.96.1.
- Adjusted pnpm-lock.yaml to reflect updated package versions and configurations.
- Added pnpm-workspace.yaml for workspace configuration.
- Updated README to reflect the new pnpm version and performance metrics.

milosh86 commented on README.md:
```diff
@@ -30,24 +30,40 @@ Before setting up the project, ensure you have the following installed:
 
+**Note**: pnpm will automatically use the correct Node.js version (22.20.0) as configured in the project.
+
+## Why pnpm?
```
I wouldn't put this justification into README. We can create a separate "decision record" document for this.

tyhonchik replied:
agree
done

thekidnamedkd commented on next.config.mjs:
```diff
@@ -98,7 +98,7 @@ const nextConfig = {
                 source: '/api/:path*',
                 headers: [
-                    { key: 'Access-Control-Allow-Origin', value: process.env.NEXT_PUBLIC_API_ALLOWED_DOMAIN },
+                    { key: 'Access-Control-Allow-Origin', value: process.env.NEXT_PUBLIC_API_ALLOWED_DOMAIN || '*' },
```
Is this change necessary for pnpm?

tyhonchik replied:
nice catch
I double-checked and it was mistake in CI job
I fixed it
thanks

thekidnamedkd commented on package.json:
```diff
@@ -104,8 +109,13 @@
+    "overrides": {
+      "@reown/appkit": "^1.8.10"
+    },
+    "packageExtensions": {}
```
is this empty packageExtensions object necessary?

tyhonchik replied:
no, it isn't
removed it

thekidnamedkd reviewed (dismissed):
This looks great too. Nice and fast, zero friction to change over on my side. ✨

tyhonchik committed:
chore(APP-207): fix PR issues
